File: app/src/main/java/com/myapp/calingaapp/UserProfile.kt
package com.myapp.calingaapp

import com.google.firebase.Timestamp

data class UserProfile(
    val userId: String = "",
    val name: String = "",
    val age: Int? = null,
    val address: String = "",
    val latitude: Double = 0.0,
    val longitude: Double = 0.0,
    val bio: String = "",
    val profilePhotoUrl: String = "",
    val specialties: List<String> = emptyList(),
    val caregiverTier: String = "",
    val isApproved: Boolean = false,
    val isActive: Boolean = false,
    val documents: List<String> = emptyList(),
    val createdAt: Timestamp = Timestamp.now()
) {
    // Empty constructor required for Firestore
    constructor() : this("", "", null, "", 0.0, 0.0, "", "", emptyList(), "", false, false, emptyList(), Timestamp.now())
    
    // Convert to Map for Firestore
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "userId" to userId,
            "name" to name,
            "age" to age,
            "address" to address,
            "latitude" to latitude,
            "longitude" to longitude,
            "bio" to bio,
            "profilePhotoUrl" to profilePhotoUrl,
            "specialties" to specialties,
            "caregiverTier" to caregiverTier,
            "isApproved" to isApproved,
            "isActive" to isActive,
            "documents" to documents,
            "createdAt" to createdAt
        )
    }
}

File: calinga_flutter_app/lib/models/user_profile.dart
class UserProfile {
  final String userId;
  final String name;
  final int age;
  final String address;
  final String email;
  final String photoUrl;
  final String userType;
  final String phoneNumber;
  final String medicalConditions;
  final String emergencyContact;

  UserProfile({
    required this.userId,
    required this.name,
    required this.age,
    required this.address,
    required this.email,
    required this.photoUrl,
    required this.userType,
    required this.phoneNumber,
    required this.medicalConditions,
    required this.emergencyContact,
  });

  // Empty constructor for Firestore
  UserProfile.empty()
    : userId = '',
      name = '',
      age = 0,
      address = '',
      email = '',
      photoUrl = '',
      userType = '',
      phoneNumber = '',
      medicalConditions = '',
      emergencyContact = '';

  // From Firestore document
  factory UserProfile.fromMap(Map<String, dynamic> map) {
    return UserProfile(
      userId: map['userId'] ?? '',
      name: map['name'] ?? '',
      age: map['age'] ?? 0,
      address: map['address'] ?? '',
      email: map['email'] ?? '',
      photoUrl: map['photoUrl'] ?? '',
      userType: map['userType'] ?? '',
      phoneNumber: map['phoneNumber'] ?? '',
      medicalConditions: map['medicalConditions'] ?? '',
      emergencyContact: map['emergencyContact'] ?? '',
    );
  }

  // To Firestore document
  Map<String, dynamic> toMap() {
    return {
      'userId': userId,
      'name': name,
      'age': age,
      'address': address,
      'email': email,
      'photoUrl': photoUrl,
      'userType': userType,
      'phoneNumber': phoneNumber,
      'medicalConditions': medicalConditions,
      'emergencyContact': emergencyContact,
    };
  }

  UserProfile copyWith({
    String? userId,
    String? name,
    int? age,
    String? address,
    String? email,
    String? photoUrl,
    String? userType,
    String? phoneNumber,
    String? medicalConditions,
    String? emergencyContact,
  }) {
    return UserProfile(
      userId: userId ?? this.userId,
      name: name ?? this.name,
      age: age ?? this.age,
      address: address ?? this.address,
      email: email ?? this.email,
      photoUrl: photoUrl ?? this.photoUrl,
      userType: userType ?? this.userType,
      phoneNumber: phoneNumber ?? this.phoneNumber,
      medicalConditions: medicalConditions ?? this.medicalConditions,
      emergencyContact: emergencyContact ?? this.emergencyContact,
    );
  }
}
